spec.bs

```diff
@@ -628,11 +654,12 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
   [=auction config/all buyer experiment group id=].
 1. Let |allBuyersGroupLimit| be |auctionConfig|'s
   [=auction config/all buyers group limit=].  
+1. Wait until |auctionConfig|'s [=auction config/pending promise count=] is 0.
 1. Let |auctionSignals| be |auctionConfig|'s [=auction config/auction signals=].
-1. If |auctionSignals| is a {{Promise}}:
-  1. [=Upon fulfillment=] of |auctionSignals| with |result| set |auctionSignals| to the result
-
-    of [=serializing a JavaScript value to a JSON string=], given |result|.
+1. [=Assert=] |auctionSignals| is not a {{Promise}}.
+1. If |auctionSignals| is failure, return failure.
+1. [=Assert=] |auctionConfig|'s [=auction config/seller signals=] is not a {{Promise}}.
+1. If |auctionConfig|'s [=auction config/seller signals=] is failure, return failure.

```

Comment:
If you wanted you could combine all of these two steps:
 1. One that assert all members are not promises; and
 2. Another that returns failure if any of them are set to failure

Reply:
Done.